Sources/PrimerSDK/Classes/Data Models/ClientSession.swift

```diff
@@ -42,31 +42,51 @@ public class ClientSession: Codable {
     
     public class Action: NSObject, Encodable {
         
-        static func unselectPaymentMethod(resumeHandler: ResumeHandlerProtocol?) {
-            DispatchQueue.main.async {
-                let actions: [ClientSession.Action] = [ClientSession.Action(type: "UNSELECT_PAYMENT_METHOD", params: nil)]
-                PrimerDelegateProxy.onClientSessionActions(actions, resumeHandler: resumeHandler)
+        static func unselectPaymentMethod() -> Promise<Void> {
+            return Promise { seal in
+                if Primer.shared.flow.internalSessionFlow.vaulted {
+                    seal.fulfill()
+                } else {
+                    let actions: [ClientSession.Action] = [ClientSession.Action(type: "UNSELECT_PAYMENT_METHOD", params: nil)]
+                    firstly {
+                        requestClientSessionWithActions(actions)
+                    }
+                    .done {
+                        seal.fulfill()
+                    }
+                    .catch { error in
+                        seal.reject(error)
+                    }
+                }
             }
         }
         
-        static func selectPaymentMethod(resumeHandler: ResumeHandlerProtocol, withParameters parameters: [String: Any]) {
-            DispatchQueue.main.async {
-                let actions: [ClientSession.Action] = [ClientSession.Action(type: "SELECT_PAYMENT_METHOD", params: parameters)]
-                PrimerDelegateProxy.onClientSessionActions(actions, resumeHandler: resumeHandler)
+        static func selectPaymentMethodWithParameters(_ parameters: [String: Any]) -> Promise<Void> {
+            return Promise { seal in
+                if Primer.shared.flow.internalSessionFlow.vaulted {
+                    seal.fulfill()
+                } else {
+                    let actions: [ClientSession.Action] = [ClientSession.Action(type: "SELECT_PAYMENT_METHOD", params: parameters)]
```

Comment:
You have to check whether it's the manual flow or not, if it is, just fullfil.

Reply:
Same for `selectPaymentMethod`

Reply:
Addressed for both ✅